romancal/refpix/refpix_step.py

```diff
@@ -0,0 +1,55 @@
+import logging
+
+import roman_datamodels as rdm
+
+from romancal.refpix import refpix
+from romancal.stpipe import RomanStep
+
+__all__ = ["RefPixStep"]
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+class RefPixStep(RomanStep):
+    """
+    RefpixStep: Module for correcting the the science data using the reference
+        pixels
+    """
+
+    spec = """
+    remove_offset = boolean(default=True) # Turn on or off removing the data offset
+    # prior to the reference pixel correction, then returning the offset afterwords.
+    remove_trends = boolean(default=True) # Turn on or off removing the boundary
+    # linear trends
+    cosine_interpolate = boolean(default=True) # Turn on or off the cosine
+    # interpolation of the reference pixels
+    fft_interpolate = boolean(default=False) # Turn on or off the FFT interpolation
+    # of the reference pixel padded values.
+    """
+
+    reference_file_types = ["refpix"]
+
+    def process(self, input):
+        """
+        Perform the reference pixel correction
+        """
+
+        # open the input data model
+        log.debug(f"Opening the science data: {input}")
+        with rdm.open(input) as datamodel:
+            # Get the reference file
+            ref_file = self.get_reference_file(datamodel, "refpix")
+
+            log.debug(f"Opening the reference file: {ref_file}")
+            with rdm.open(ref_file) as refs:
+                # Run the correction
+                log.debug("Running the reference pixel correction")
+                return refpix.run_steps(
+                    datamodel,
+                    refs,
+                    self.remove_offset,
+                    self.remove_trends,
+                    self.cosine_interpolate,
+                    self.fft_interpolate,
+                )
```

Comment:
When I run the step on the command line using
```
strun romancal.refpix.RefPixStep r0000201001001001003_01101_0001_WFI16_dqinit.asdf --override_refpix refpix_reference.asdf
```
I get  a traceback ending in 
```
...
  File "/Users/dencheva/opt/anaconda3/envs/rb10/lib/python3.9/site-packages/asdf/generic_io.py", line 382, in seek
    result = self._fd.seek(offset, whence)
ValueError: seek of closed file
```

I think the reason is that the datamodel is returned within a couple of context managers which close the file before exiting the function. We need the `process` function to return the datamodel so it can be passed to the next step in the pipeline, or saved to file by the user.

Reply:
The issue you are seeing is from the outer context manager not the because of the use of nested managers.

I updated the model in place to avoid copying input model. Would you rather I just copy the input model and return the copy?